src/main/java/io/cdap/plugin/jira/source/streaming/JiraInputDStream.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.jira.source.streaming;
+
+import com.atlassian.jira.rest.client.api.domain.Issue;
+import com.atlassian.jira.rest.client.api.domain.SearchResult;
+import com.google.common.base.Throwables;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.plugin.jira.source.common.IssueToRecordTransformer;
+import io.cdap.plugin.jira.source.common.JiraClient;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.rdd.RDD;
+import org.apache.spark.streaming.StreamingContext;
+import org.apache.spark.streaming.Time;
+import org.apache.spark.streaming.dstream.InputDStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.reflect.ClassTag;
+
+import java.time.ZonedDateTime;
+import java.util.LinkedList;
+
+/**
+ * Iterates over the jira issues and fills Spark RDD with structured records from them.
+ *
+ * Note: Fields of this class get checkpointed, which means that their value is restored if pipeline is re-run
+ * after a stop.
+ */
+public class JiraInputDStream extends InputDStream<StructuredRecord> {
```

Comment:
why not Receiver API? It is more Java-friendly.

Reply:
because checkpointing doesn't work with it.